src/wp-content/themes/twentytwentyone/assets/js/interactivity.js

```diff
@@ -0,0 +1,206 @@
+            elements.forEach( ( subMenuToggle ) => {
+                if ( ctx.activeSubmenu === subMenuToggle ) {
+                    subMenuToggle.setAttribute( 'aria-expanded', 'true' );
+                } else {
+                    subMenuToggle.setAttribute( 'aria-expanded', 'false' );
+                }
+            } );
+        },
+
+        makeIframesResponsive: () => {
+            const { ref } = getElement();
+
+            ref.querySelectorAll( 'iframe' ).forEach( function( iframe ) {
+                // Only continue if the iframe has a width & height defined.
+                if ( iframe.width && iframe.height ) {
+                    // Calculate the proportion/ratio based on the width & height.
+                    proportion = parseFloat( iframe.width ) / parseFloat( iframe.height );
+                    // Get the parent element's width.
+                    parentWidth = parseFloat( window.getComputedStyle( iframe.parentElement, null ).width.replace( 'px', '' ) );
+                    // Set the max-width & height.
+                    iframe.style.maxWidth = '100%';
+                    iframe.style.maxHeight = Math.round( parentWidth / proportion ).toString() + 'px';
+                }
+            } );
+        },
+
+        updateWindowWidthOnResize: () => {
+            // The following may be needed here since we can't use `data-wp-on--resize`?
+            const refreshWidth = () => {
+                state.windowWidth = window.innerWidth;
+            }
+            window.onresize = refreshWidth;
+        },
+
+        initDarkMode: () => {
+            let isDarkMode = window.matchMedia( '(prefers-color-scheme: dark)' ).matches;
+
+            if ( 'yes' === window.localStorage.getItem( 'twentytwentyoneDarkMode' ) ) {
+                isDarkMode = true;
+            } else if ( 'no' === window.localStorage.getItem( 'twentytwentyoneDarkMode' ) ) {
+                isDarkMode = false;
+            }
+
+            state.isDarkMode = isDarkMode;
+
+            // The following may be needed here since we can't use `data-wp-on--scroll`?
+            const checkScroll = () => {
+                const currentScroll = window.scrollY || document.documentElement.scrollTop;
+                if (
+                    currentScroll + ( window.innerHeight * 1.5 ) > document.body.clientHeight ||
+                    currentScroll < state.prevScroll
+                ) {
+                    state.isDarkModeTogglerHidden = false;
+                } else if ( currentScroll > state.prevScroll && 250 < currentScroll ) {
+                    state.isDarkModeTogglerHidden = true;
+                }
+                state.prevScroll = currentScroll;
+            }
+            window.addEventListener( 'scroll', checkScroll );
+        },
+
+        refreshHtmlElementDarkMode: () => {
+            // This hack may be needed since the HTML element cannot be controlled with the API attributes?
```

Comment:
`<html>` is not supported as an interactivity target, so this is the exception where mutating the DOM manually is correct.

Oh, and doing so reactively with a `data-wp-watch` like you're doing here is the best possible way to do so because it will always be in sync, no matter where `state.isDarkMode` is changed 🙂